facebook-core/src/main/java/com/facebook/internal/Utility.java

```diff
@@ -1186,4 +1187,25 @@ public static String generateRandomString(int length) {
         Random r = new Random();
         return new BigInteger(length * 5, r).toString(32);
     }
+
+    /*
+     * There is a bug on Android O that excludes the dialog's view hierarchy from the
+     * ViewStructure used by Autofill because the window token is lost when the dialog
+     * is resized, hence the token needs to be saved dialog is attached to a window and restored
+     * when the dialog attributes change after it is resized.
+     */
+    public static boolean mustFixWindowParamsForAutofill(Context context) {
+        // TODO: once this bug is fixed on Android P, checks for version here as well
```

Comment:
ETA?

Reply:
I don't know the exact dates P will be launched, but based on this year schedule for O, most likely the first developer preview will be available on Q1/2018, which the final launch on Q3/2018